treeherder/auth/backends.py

```diff
@@ -49,12 +49,23 @@ def _get_email(self, result):
         For more info on scopes:
         https://docs.taskcluster.net/manual/3rdparty#authenticating-with-scopes
         """
+        # Try finding the email in the mozilla-user scope
         email = self._get_scope_value(result["scopes"], "assume:mozilla-user:")
 
         if email and re.search(r'.+@.+', email):
             return email
+        else:
+            # Try finding the email in the clientId.
+            # Credit for regex to http://emailregex.com/ Python section
+            match = re.search(
+                r"([a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+)",
```

Comment:
Ran into Dustin and we hashed out the right approach to take for both the LDAP and the Email login paths:

LDAP login:

1. extract email from the client id.
2. ask if that email exists in the scope of mozilla-user
3. YES: find existing user and allow sheriff, migrate username to clientid
4. NO: then create a new user with that clientid

Email login:

1. search for either clientId or hashed_clientid[:30]
2. if found, login
3. if not found, create